x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -776,7 +792,7 @@ export class Authenticator {
     sessionValue,
     skipAuditEvent,
   }: InvalidateSessionValueParams) {
-    if (sessionValue && !skipAuditEvent) {
+    if (isSessionAuthenticated(sessionValue) && !skipAuditEvent) {
```

Comment:
I just saw the type errors related to this 🙄 Sometimes the right code is harder to write in TS 😅 

I think the way to fix it, is something like this:

```ts
function isSessionAuthenticated(sessionValue?: Readonly<SessionValue> | null): sessionValue is SessionValue {
  return !!sessionValue?.username;
}
```

But this will not be 100% correct as sessionValue is not only set but also authenticated. So it will give some other TS errors if you copy this directly. But it's along those lines.

Reply:
I think I found the full solution:

```ts
type WithRequiredProperty<Type, Key extends keyof Type> = Type & {
  [Property in Key]-?: Type[Property];
};

function isSessionAuthenticated(sessionValue?: Readonly<SessionValue> | null): sessionValue is WithRequiredProperty<SessionValue, 'username'> {
  return !!sessionValue?.username;
}
```

Reply:
Oh shit, sorry, I didn't see this before merging. 

Nice one, I tried the first solution as well but it didn't infer the correct type for the false return value. 

Opened a follow up here:  #139978